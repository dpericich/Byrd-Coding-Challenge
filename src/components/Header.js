import React from "react";
import "./header.css";
import mainLogo from "../assets/logo-main.png";
import whiteLogo from "../assets/logo-white.png";

const Header = ({ color }) => {
  return (
    <div className="header">
      <div className="header__social-icons">
        <ion-icon name="logo-facebook"></ion-icon>
        <ion-icon name="logo-twitter"></ion-icon>
      </div>
      <img
        src={color === "orange" ? mainLogo : whiteLogo}
        alt="logo for header"
        className="header__logo"
      />
      <div className="header__dropdown-icon">
        <ion-icon name="reorder-two-outline"></ion-icon>
      </div>
    </div>
  );
};

export default Header;
