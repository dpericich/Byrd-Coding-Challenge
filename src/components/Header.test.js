import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Header from "./Header";

describe("<Header />", () => {
  test("it renders the header", () => {
    const component = shallow(<Header />);

    expect(toJSON(component)).toMatchSnapshot();
  });

  test("it renders with logo color argument prop", () => {
    const color = "orange";

    const component = shallow(<Header color={color} />);

    expect(toJSON(component)).toMatchSnapshot();
  });
});
