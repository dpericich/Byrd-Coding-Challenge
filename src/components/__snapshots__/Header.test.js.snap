// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> it renders the header 1`] = `
<div
  className="header"
>
  <div
    className="header__social-icons"
  >
    <ion-icon
      name="logo-facebook"
    />
    <ion-icon
      name="logo-twitter"
    />
  </div>
  <img
    alt="logo for header"
    className="header__logo"
    src="logo-white.png"
  />
  <div
    className="header__dropdown-icon"
  >
    <ion-icon
      name="reorder-two-outline"
    />
  </div>
</div>
`;

exports[`<Header /> it renders with logo color argument prop 1`] = `
<div
  className="header"
>
  <div
    className="header__social-icons"
  >
    <ion-icon
      name="logo-facebook"
    />
    <ion-icon
      name="logo-twitter"
    />
  </div>
  <img
    alt="logo for header"
    className="header__logo"
    src="logo-main.png"
  />
  <div
    className="header__dropdown-icon"
  >
    <ion-icon
      name="reorder-two-outline"
    />
  </div>
</div>
`;
